pkg/machine/ports_unix.go

```diff
@@ -0,0 +1,25 @@
+//go:build darwin || dragonfly || freebsd || linux || netbsd || openbsd
+// +build darwin dragonfly freebsd linux netbsd openbsd
+
+package machine
+
+import (
+	"net"
+	"syscall"
+)
+
+func getPortCheckListenConfig() *net.ListenConfig {
+	return &net.ListenConfig{
+		Control: func(network, address string, c syscall.RawConn) (cerr error) {
+			if err := c.Control(func(fd uintptr) {
+				cerr = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_REUSEADDR, 0)
```

Comment:
why SO_REUSEADDR? The seems to be the opposite of what you are trying to do. With SO_REUSEADDR it would allow you to bind the same port if both sockets have SO_REUSEADDR set so you actually do not check for a free port with that.

I really do not understand why this function is needed at all and there are no comments to explain it either.

Reply:
@Luap99 it’s explicitly disabling SO_REUSEADDR as a precaution, since to would break detection, although I should just remove that line, since its unlikely to be ever needed: the only OS I knew that ever defaulted that way was very old versions of windows. The main reason for the function is to set linger with a 0 timeout in the rare case something tries to connect and gets stuck in TIME_WAIT preventing the real later binding by a different process. Although I don’t know what I was thinking and didn’t set this properly. I will fix that

Reply:
ok please add comments to this code why it explicitly sets these

Reply:
sure will do